jetty-util-ajax/src/main/java/org/eclipse/jetty/util/ajax/AsyncJSON.java

```diff
@@ -0,0 +1,1205 @@
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.util.ajax;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.eclipse.jetty.util.ArrayTernaryTrie;
+import org.eclipse.jetty.util.BufferUtil;
+import org.eclipse.jetty.util.Loader;
+import org.eclipse.jetty.util.Trie;
+import org.eclipse.jetty.util.TypeUtil;
+import org.eclipse.jetty.util.Utf8StringBuilder;
+
+public class AsyncJSON
+{
+    public static class Factory
+    {
+        private Trie<String> cache;
+        private Map<String, JSON.Convertor> convertors;
+        private boolean detailedParseException;
+
+        public boolean isDetailedParseException()
+        {
+            return detailedParseException;
+        }
+
+        public void setDetailedParseException(boolean detailedParseException)
+        {
+            this.detailedParseException = detailedParseException;
+        }
+
+        public void cache(String value)
+        {
+            if (cache == null)
+                cache = new ArrayTernaryTrie.Growing<>(false, 64, 64);
+            cache.put("\"" + value + "\"", value);
+        }
+
+        public String cached(ByteBuffer buffer)
+        {
+            if (cache != null)
+            {
+                String result = cache.getBest(buffer, 0, buffer.remaining());
+                if (result != null)
+                {
+                    buffer.position(buffer.position() + result.length() + 2);
+                    return result;
+                }
+            }
+            return null;
+        }
+
+        public AsyncJSON newAsyncJSON()
+        {
+            return new AsyncJSON(this);
+        }
+
+        public void putConvertor(String className, JSON.Convertor convertor)
+        {
+            if (convertors == null)
+                convertors = new ConcurrentHashMap<>();
+            convertors.put(className, convertor);
+        }
+
+        public JSON.Convertor removeConvertor(String className)
+        {
+            if (convertors != null)
+                return convertors.remove(className);
+            return null;
+        }
+    }
+
+    private static final Object UNSET = new Object();
+
+    private final Factory factory;
+    private final FrameStack stack = new FrameStack();
```

Comment:
You probably want to initialize the `stack` in the constructor, or remove the initializations of `stringBuilder` and `numberBuilder` from the constructor.

Reply:
Okay